doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -4,99 +4,82 @@
+   tries to apply one of them.  Lower cost tactics are tried before higher-cost
+   tactics.  This process is recursively applied to the generated subgoals.
 
-   .. tacv:: auto with *
+   If no `with` clause is givesn, :tacn:`auto` only uses the hypotheses of the
+   current goal and the hints of the database named ``core``.
 
-      Uses all existing hint databases. Using this variant is highly discouraged
-      in finished scripts since it is both slower and less robust than the variant
-      where the required databases are explicitly listed.
+   .. warning::
+
+      :tacn:`auto` uses a weaker version of :tacn:`apply` that is closer to
+      :tacn:`simple apply` so it is expected that sometimes :tacn:`auto` will
+      fail even if applying manually one of the hints would succeed.
 
    .. seealso::
       :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>` for the list of
       pre-defined databases and the way to create or extend a database.
 
-   .. tacv:: auto using {+ @qualid__i} {? with {+ @ident } }
-
-      Uses lemmas :n:`@qualid__i` in addition to hints. If :n:`@qualid` is an
-      inductive type, it is the collection of its constructors which are added
-      as hints.
-
-      .. note::
-
-         The hints passed through the `using` clause are used in the same
-         way as if they were passed through a hint database. Consequently,
-         they use a weaker version of :tacn:`apply` and :n:`auto using @qualid`
-         may fail where :n:`apply @qualid` succeeds.
-
-         Given that this can be seen as counter-intuitive, it could be useful
-         to have an option to use full-blown :tacn:`apply` for lemmas passed
-         through the `using` clause. Contributions welcome!
-
-   .. tacv:: info_auto
+   .. tacn:: info_auto {? @int_or_var } {? @auto_using } {? @hintbases }
 
       Behaves like :tacn:`auto` but shows the tactics it uses to solve the goal. This
       variant is very useful for getting a better understanding of automation, or
       to know what lemmas/assumptions were used.
 
-   .. tacv:: debug auto
-      :name: debug auto
+   .. tacn:: debug auto {? @int_or_var } {? @auto_using } {? @hintbases }
 
       Behaves like :tacn:`auto` but shows the tactics it tries to solve the goal,
       including failing paths.
 
-   .. tacv:: {? info_}auto {? @natural} {? using {+ @qualid}} {? with {+ @ident}}
-
-      This is the most general form, combining the various options.
+.. tacn:: trivial {? @auto_using } {? @hintbases }
+          debug trivial {? @auto_using } {? @hintbases }
+          info_trivial {? @auto_using } {? @hintbases }
 
-.. tacv:: trivial
-   :name: trivial
-
-   This tactic is a restriction of :tacn:`auto` that is not recursive
-   and tries only hints that cost `0`. Typically it solves trivial
+   Like :tacn:`auto`, but is not recursive
+   and only tries hints with zero cost. Typically it solves trivial
    equalities like :g:`X=X`.
```

Comment:
I don't think it is a good example. We could instead say that it is typically used to solve goals for which a lemma is already available in the hint database that is used.